akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -546,6 +546,7 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
           updateLastSequenceNr(p)
           try {
             peekApplyHandler(p.payload)
+            onPersistSuccess(p.payload)
```

Comment:
IMHO it should be handled in the same way the callback is handled. 

The only omission is I guess the functionality inside `peekApplyHandler` to flush any added persists. Am I missing something else?